flask/app/families.py

```diff
@@ -23,3 +25,62 @@ def families_list():
     ]
 
     return jsonify(families)
+
+@app.route("/api/families/<int:id>", methods=["GET"])
+@login_required
+def families_by_id(id: int):
+    family = models.Family.query.filter_by(family_id = id).options(
+        joinedload(models.Family.participants)\
+            .joinedload(models.Participant.tissue_samples)\
+                .joinedload(models.TissueSample.datasets)
+    ).first_or_404()
+
+
+    families = [
+        { 
+            **asdict(family),
+            "participants": [
+            {
+                **asdict(participants),
+                "tissue_samples": [
+                        {
+                            **asdict(tissue_samples),
+                            "datasets": [
+                                {
+                                    **asdict(dataset)
+                                }
+                            
+                                for dataset in tissue_samples.datasets
+                            ]
+                        }
+                        for tissue_samples in participants.tissue_samples
+                    ]
+                }
+            for participants in family.participants
+            ]
+        }
+    ]
+    return jsonify(families)
+
+
+@app.route("/api/families/<int:id>", methods = ["DELETE"])
+@login_required
+@check_admin
+def delete_families(id: int):
+    family = models.Family.query.filter_by(family_id = id).options(
+        joinedload(models.Family.participants)
+    )
+
+    fam_entity = family.first_or_404()
+
+    if len(fam_entity.participants) == 0:
+        try:
+            family.delete()
+            db.session.commit()
+            return 'Deletion successful', 200
```

Comment:
Should this be code 204? (I don't know which is more correct; I've used 204 in my DELETE endpoints because that's what was in the DELETE users endpoint- it might be good to standardize either way)